sdk/aqueduct/aqueduct_client.py

```diff
@@ -365,8 +367,14 @@ def trigger(
             InternalServerError:
                 An unexpected error occurred within the Aqueduct cluster.
         """
-        # TODO(ENG-1144): If there the provided parameters dict is not valid, throw an error
-        #  earlier, before getting to execution.
+        if parameters is not None:
+            flow = self.flow(flow_id)
+            runs = flow.list_runs(limit=1)
+            if len(runs) == 0:
+                raise InvalidUserActionException(
+                    "Cannot trigger a workflow that hasn't already run at least once."
+                )
```

Comment:
@saurav-c @vsreekanti I could also try and fetch the latest workflow dag here, even if there weren't any workflow runs, but the implementation of that was a little messy. It seemed logical to simply disallow `trigger()` until at least one run exists, but lemme know if thats not good.

Reply:
I'm actually not 100% sure how triggering works with Airflow -- @saurav-c, is there an API hook to trigger a workflow that has been registered, or can it only be on a schedule?

I'm asking because I wonder if it makes sense to have system-specific trigger checks.

Reply:
@vsreekanti There is an API hook to trigger an Airflow dag if its been registered.